deepdrivemd/aggregation/stream/aggregator.py

```diff
@@ -0,0 +1,207 @@
+import numpy as np
+from deepdrivemd.aggregation.stream.config import StreamAggregation
+from deepdrivemd.utils import Timer, timer, parse_args, intarray2hash
+import time
+import os
+import sys
+import adios2
+import math
+import queue
+import subprocess
+import itertools
+
+
+def find_input(cfg: StreamAggregation):
+    """
+    Find and return simulation adios streams.
+    Wait until those are created by simulations.
+    Returns the list of *.sst files associated with the corresponding adios streams.
+    """
+    while True:
+        bpfiles = list(map(str, list(cfg.experiment_directory.glob("*/*/*/md.bp*"))))
+        if len(bpfiles) == cfg.n_sim:
+            break
+        print("In find_input: waiting for input")
+        time.sleep(cfg.sleeptime_bpfiles)
+
+    bpfiles.sort()
+    return bpfiles
+
+
+def connect_to_input(cfg: StreamAggregation, bpfiles):
+    """
+    Open adios streams for reading and return a dictionary: key - simulation task id,
+    value - tuple of the corresponding adios objects
+    """
+    connections = {}
+    bp_slice = math.ceil(cfg.n_sim / cfg.num_tasks)
+    print("bp_slice = ", bp_slice)
+    i = 0
+    for bp in bpfiles:
+        bp = bp.replace(".sst", "")
+        dir = os.path.dirname(bp)
+        task_md = os.path.basename(dir)
+        taskid_md = int(task_md.replace("task", ""))
+        adios_md = dir + "/adios.xml"
+
+        taskid_agg = cfg.task_idx
```

Comment:
This variable is never mutated. Can get rid of it, and simply use `cfg.task_idx` below.

Reply:
Done